Userland/Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -1490,6 +1490,17 @@ WebIDL::ExceptionOr<String> Node::serialize_fragment(DOMParsing::RequireWellForm
         return HTML::HTMLParser::serialize_html_fragment(*this, HTML::HTMLParser::SerializableShadowRoots::No, {}, fragment_serialization_mode);
 
     // 3. Return the XML serialization of node given require well-formed.
+    // AD-HOC: XML serialization algorithm returns the "outer" XML serialization of the node.
+    //         For inner, concatenate the serialization of all children.
+    if (fragment_serialization_mode == FragmentSerializationMode::Inner) {
```

Comment:
Is this maybe a spec bug? The [XML serializing an Element node](https://w3c.github.io/DOM-Parsing/#xml-serializing-an-element-node) algorithm is (almost) directly called from the `innerHTML` getter even though it serializes the outer XML...

Reply:
The spec does indeed expect the fragment serializing algorithm to return the contents rather than the node itself (see <https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#the-outerhtml-property>).
I'd call that a spec bug.

Reply:
Please consider raising a spec issue at https://github.com/whatwg/html/issues/new/choose

Reply:
...and linking the spec bug here. We've got that pattern in a few places throughout LibWeb already.

Reply:
https://github.com/whatwg/html/issues/10478